examples/driving_in_traffic/run.py

```diff
@@ -57,46 +58,24 @@ def main(args):
     tf.config.run_functions_eagerly(not config_dv2.jit)
 
     # Setup GPU.
-    gpus = tf.config.list_physical_devices("GPU")
-    if gpus:
-        try:
-            for gpu in gpus:
-                tf.config.experimental.set_memory_growth(gpu, True)
-            logical_gpus = tf.config.list_logical_devices("GPU")
-            print(len(gpus), "Physical GPUs,", len(logical_gpus), "Logical GPUs")
-        except RuntimeError as e:
-            print(e)
-    else:
-        warnings.warn(
-            f"Not configured to use GPU or GPU not available.",
-            ResourceWarning,
-        )
+    _setup_gpu()
 
     # Train or evaluate.
-    if config_env["mode"] == "train":
-        if not config_env["logdir"]:
-            # Begin training from scratch.
-            time = datetime.now().strftime("%Y_%m_%d_%H_%M")
-            logdir = pathlib.Path(__file__).absolute().parents[0] / "logs" / time
-        else:
-            # Begin training from a pretrained model.
-            logdir = config_env["logdir"]
-        config_dv2 = config_dv2.update(
-            {
-                "logdir": logdir,
-            }
-        )
+    if config_env["mode"] == "train" and not args.logdir:
+        # Train from scratch.
+        time = datetime.now().strftime("%Y_%m_%d_%H_%M")
+        logdir = pathlib.Path(__file__).absolute().parents[0] / "logs" / time
+    elif config_env["mode"] == "train" and args.logdir:
+        # Begin training from a pretrained model.
+        logdir = args.logdir
     elif config_env["mode"] == "evaluate":
-        config_dv2 = config_dv2.update(
-            {
-                "logdir": config_env["logdir"],
-                "eval_eps": 1e8,
-            }
-        )
+        logdir = args.logdir
+        config_dv2 = config_dv2.update({"eval_eps": 1e8})
     else:
         raise KeyError(
             f'Expected \'train\' or \'evaluate\', but got {config_env["mode"]}.'
         )
+    config_dv2 = config_dv2.update({"logdir": logdir})
```

Comment:
Why is the `logdir` path stored in the configuration file if it is being passed by user and not being used by the config file anywhere?

Reply:
It is being used later in the `run` function.
https://github.com/huawei-noah/SMARTS/blob/a3a32ddb96e0d4cd41d69e7de7c8d559f2dab585/examples/driving_in_traffic/run.py#L126